File: benchmarks/strings/strings.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	text := "the quick brown fox jumps over the lazy dog"
	checksum := 0
	for i := 0; i < 60000; i++ {
		joined := strings.Join(strings.Split(text, " "), "-")
		checksum += strings.Index(joined, "fox")
		checksum += len(strings.ToUpper(joined))
		replaced := strings.Replace(joined, "quick", "slow", 1)
		checksum += strings.LastIndex(replaced, "o")
	}
	fmt.Println(checksum)
}

File: benchmarks/strings/strings.rs
fn main() {
    let text = "the quick brown fox jumps over the lazy dog";
    let mut checksum: i64 = 0;
    for _ in 0..60000 {
        let joined = text.split(' ').collect::<Vec<_>>().join("-");
        checksum += joined.find("fox").map(|i| i as i64).unwrap_or(-1);
        checksum += joined.to_uppercase().len() as i64;
        let replaced = joined.replacen("quick", "slow", 1);
        checksum += replaced.rfind('o').map(|i| i as i64).unwrap_or(-1);
    }
    println!("{checksum}");
}
